app/controllers/users_controller.rb

```diff
@@ -147,6 +149,18 @@ def reset_user_password
     redirect_to users_path
   end
 
+  def emails_csv
+    emails = params['team'] == TEAMS::ALL ? get_all_emails_for_users_in_teams(Team.where(team_type: [TEAMS::IDP, TEAMS::RP])) : get_all_emails_for_users_in_teams(Team.where(team_type: params['team']))
```

Comment:
I would define a scope in Team

```
scope :retrieve, ->(team) { where(team_id: team) }
```

```
emails =  get_all_emails_for_users_in_teams(Team.retrieve(params['team'] == TEAMS::ALL ? [TEAMS::IDP, TEAMS::RP] : params['team'])) 
```

Reply:
Good idea 👍🏽 I remember doing something similar with the components